src/frequenz/sdk/actor/power_distributing/_component_managers/_battery_manager.py

```diff
@@ -649,21 +654,22 @@ def _parse_result(
             battery_ids = self._inv_bats_map[inverter_id]
             try:
                 aws.result()
-            except grpc.aio.AioRpcError as err:
+            except OperationOutOfRange as err:
+                failed_power += distribution[inverter_id]
+                failed_batteries = failed_batteries.union(battery_ids)
+                _logger.debug(
+                    "Set power for battery %s failed due to out of range error: %s",
+                    battery_ids,
+                    err,
+                )
+            except ClientError as err:
                 failed_power += distribution[inverter_id]
                 failed_batteries = failed_batteries.union(battery_ids)
-                if err.code() == grpc.StatusCode.OUT_OF_RANGE:
-                    _logger.debug(
-                        "Set power for battery %s failed, error %s",
-                        battery_ids,
-                        str(err),
-                    )
-                else:
-                    _logger.warning(
-                        "Set power for battery %s failed, error %s. Mark it as broken.",
-                        battery_ids,
-                        str(err),
-                    )
+                _logger.warning(
+                    "Set power for battery %s failed, mark it as broken. Error: %s",
+                    battery_ids,
+                    err,
+                )
```

Comment:
I think ignoring a failed precondition could be fine, it just means there something else that needs to be done before retrying could succeed:

> The operation was rejected because the system is not in a state required for the operation's execution. For example, the directory to be deleted is non-empty, an rmdir operation is applied to a non-directory, etc. Service implementors can use the following guidelines to decide between FAILED_PRECONDITION, ABORTED, and UNAVAILABLE: (a) Use UNAVAILABLE if the client can retry just the failing call. (b) Use ABORTED if the client should retry at a higher level (e.g., when a client-specified test-and-set fails, indicating the client should restart a read-modify-write sequence). (c) Use FAILED_PRECONDITION if the client should not retry until the system state has been explicitly fixed. E.g., if an "rmdir" fails because the directory is non-empty, FAILED_PRECONDITION should be returned since the client should not retry unless the files are deleted from the directory. 

Out of range is considered some sort of special case of failed precondition:

> There is a fair bit of overlap between FAILED_PRECONDITION and OUT_OF_RANGE. We recommend using OUT_OF_RANGE (the more specific error) when it applies so that callers who are iterating through a space can easily look for an OUT_OF_RANGE error to detect when they are done. 

I agree out of range could make more sense if the request fails because the power is out of bounds.